client/asset/btc/rpcclient.go

```diff
@@ -790,33 +791,49 @@ func (wc *rpcClient) sendToAddress(address string, value, feeRate uint64, subtra
 	return chainhash.NewHashFromStr(txid)
 }
 
+// sendTxFeeEstimator returns the fee required to send tx using the provided
+// feeRate.
 func (wc *rpcClient) estimateSendTxFee(tx *wire.MsgTx, feeRate uint64, subtract bool) (txfee uint64, err error) {
 	txBytes, err := wc.serializeTx(tx)
 	if err != nil {
 		return 0, fmt.Errorf("tx serialization error: %w", err)
 	}
+	args := anylist{hex.EncodeToString(txBytes)}
 
 	// 1e-5 = 1e-8 for satoshis * 1000 for kB.
 	feeRateOption := float64(feeRate) / 1e5
-	options := &btcjson.FundRawTransactionOpts{
-		FeeRate:    &feeRateOption,
-		ChangeType: &btcjson.ChangeTypeBech32,
-	}
-	if subtract {
-		options.SubtractFeeFromOutputs = []int{0}
-	}
-	if !wc.segwit {
-		options.ChangeType = &btcjson.ChangeTypeLegacy
+	if wc.OmitRPCOptionsArg {
+		var success bool
+		err := wc.call(methodSetTxFee, anylist{feeRateOption}, &success)
+		if err != nil {
+			return 0, fmt.Errorf("error setting transaction fee: %w", err)
+		}
+		if !success {
+			return 0, fmt.Errorf("failed to set transaction fee")
+		}
+	} else {
+		options := &btcjson.FundRawTransactionOpts{
+			FeeRate: &feeRateOption,
+		}
+		if !wc.omitAddressType {
```

Comment:
Excellent. The existing `omitAddressType` is perfectly suited AFAICT.

However, please unexport `rpcCore.OmitRPCOptionsArg`.  It is only supposed to be exported in `BTCCloneCFG`

Reply:
oops, I didn't notice.